core/trino-spi/src/main/java/io/trino/spi/block/DictionaryBlock.java

```diff
@@ -273,6 +273,12 @@ public long getLogicalSizeInBytes()
             return logicalSizeInBytes;
         }
 
+        OptionalInt dictionarySizePerPosition = dictionary.fixedSizeInBytesPerPosition();
+        if (dictionarySizePerPosition.isPresent()) {
+            logicalSizeInBytes = dictionarySizePerPosition.getAsInt() * (long) getPositionCount();
```

Comment:
inline

Reply:
`logicalSizeInBytes` is a class member variable which needs to be updated before return